test/libsolidity/semanticTests/builtins/balance_other_contract.sol

```diff
@@ -0,0 +1,25 @@
+contract Other {
+    constructor() payable {
+    }
+    function getAddress() public returns (address) {
+        return address(this);
+    }
+}
+contract ClientReceipt {
+    Other other;
+    constructor() payable {
+        other = new Other{value:500}();
+    }
+    function getAddress() public returns (address) {
+        return other.getAddress();
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor(), 2000 wei ->
+// gas legacy: 235167
```

Comment:
@mijovic Not sure why isoltest expectation update generate this `gas legacy` measurements. Any ideas? Why here but not for the other tests defined here?

Reply:
@mijovic after looking into https://app.circleci.com/pipelines/github/ethereum/solidity/15296/workflows/15367350-883a-464e-b764-0c2093dcc398/jobs/695532, I'm wondering why we see here additionally `gas legacyOptimized`.

Reply:
Both are more than 100K gas, that is why it forces these to be present. Also, you need to run `isoltest` with `--optimize` locally to trigger checks for optimized versions

Reply:
Ok, understood. Thanks!